app/src/main/java/com/taylorgirard/comicconvo/activities/IndividualMessageActivity.java

```diff
@@ -1,17 +1,213 @@
 package com.taylorgirard.comicconvo.activities;
 
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
+import android.content.Intent;
+import android.location.Location;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
 import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.TextView;
 
+import com.parse.FindCallback;
+import com.parse.FunctionCallback;
+import com.parse.Parse;
+import com.parse.ParseCloud;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+import com.parse.livequery.ParseLiveQueryClient;
+import com.parse.livequery.SubscriptionHandling;
 import com.taylorgirard.comicconvo.R;
+import com.taylorgirard.comicconvo.adapters.MessageAdapter;
+import com.taylorgirard.comicconvo.models.Message;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 
 public class IndividualMessageActivity extends AppCompatActivity {
 
+    public static final String TAG = "IndividualMessageActivity";
+
+    EditText etMessage;
+    ImageButton ibSend;
+    ImageButton ibMakePin;
+    RecyclerView rvMessages;
+    ArrayList<Message> messages;
+    MessageAdapter adapter;
+    ParseUser match;
+    ParseUser user;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_individual_message);
+        setupMessagePosting();
+        loadMessages();
+    }
+
+    @Override
+    public void onBackPressed() {
+        Intent intent = new Intent(IndividualMessageActivity.this, MainActivity.class);
```

Comment:
I don't want the back press functionality to be what it usually is because then the information will not be updated. Instead I want the user to be pushed back to the homepage so that when they reload their message list the new changes are reflected